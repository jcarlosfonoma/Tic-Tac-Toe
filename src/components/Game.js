import Board from "./Board";
import styled from "styled-components";

const GameInfo = styled.div`
  margin-left: 20px;
  background-color: #3d4250;
  border-radius: 12px;
  color: white;
`;

const Status = styled.div`
  margin-bottom: 10px;
  margin-top: 10px;
  text-align: center;
  font-weight: bold;
`;

const PLayList = styled.ol`
  list-style: none;
  padding: 0;
`;

const Game = (props) => {
  return (
    <div className="game">
      <div className="game-board">
        <Board data-testid="board-btn" squares={props.squares} onClick={(i) => props.onClick(i)} />
      </div>
      <GameInfo>
        <Status data-testid="status-div">{props.status}</Status>
        <PLayList>{props.moves}</PLayList>
      </GameInfo>
    </div>
  );
};

export default Game;
