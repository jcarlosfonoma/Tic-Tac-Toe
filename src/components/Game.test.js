import { cleanup, render } from "@testing-library/react";
import Game from "./Game";

describe("Game component", () => {
  beforeEach(cleanup);
  const squares = Array(9).fill(null);

  it("Game should take a snapshot", () => {
    const { asFragment } = render(<Game squares={squares} />);
    expect(asFragment(<Game />)).toMatchSnapshot();
  });

  it("Checks if initial status is equal to Go to game start", () => {
    const { getByTestId } = render(
      <Game squares={squares} status="Go to game start" />
    );
    expect(getByTestId("status-div")).toHaveTextContent("Go to game start");
  });
});
