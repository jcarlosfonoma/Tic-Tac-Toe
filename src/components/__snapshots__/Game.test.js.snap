// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component Game should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="game"
  >
    <div
      class="game-board"
    >
      <div>
        <div
          class="board-row"
        >
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
        </div>
        <div
          class="board-row"
        >
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
        </div>
        <div
          class="board-row"
        >
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
          <button
            class="sc-bdvvtL fPtRnW"
            data-testid="square-btn"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-gsDKAQ gKFSjR"
    >
      <div
        class="sc-dkPtRN hDBrjf"
        data-testid="status-div"
      />
      <ol
        class="sc-hKwDye fqVKqO"
      />
    </div>
  </div>
</DocumentFragment>
`;
